internal/command/migrate_to_v2/migrate_to_v2.go

```diff
@@ -455,10 +546,290 @@ func (m *v2PlatformMigrator) Migrate(ctx context.Context) (err error) {
+	app, err := m.apiClient.GetImageInfo(ctx, m.appCompact.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get image info: %w", err)
+	}
+
+	if !app.ImageVersionTrackingEnabled {
+		return errors.New("image is not eligible for automated image updates")
+	}
+
+	if !app.ImageUpgradeAvailable {
+		return nil
+	}
+
+	cI := app.ImageDetails
+	lI := app.LatestImageDetails
+
+	current := cI.FullImageRef()
+	target := lI.FullImageRef()
+
+	if cI.Version != "" {
+		current = fmt.Sprintf("%s %s", current, cI.Version)
+	}
+
+	if lI.Version != "" {
+		target = fmt.Sprintf("%s %s", target, lI.Version)
+	}
+
+	input := api.DeployImageInput{
+		AppID:    m.appCompact.Name,
+		Image:    lI.FullImageRef(),
+		Strategy: api.StringPointer("ROLLING"),
+	}
+
+	// Set the deployment strategy
+	if val := flag.GetString(ctx, "strategy"); val != "" {
+		input.Strategy = api.StringPointer(strings.ReplaceAll(strings.ToUpper(val), "-", "_"))
+	}
+
+	release, releaseCommand, err := m.apiClient.DeployImage(ctx, input)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(m.io.Out, "Release v%d created\n", release.Version)
+	if releaseCommand != nil {
+		fmt.Fprintln(m.io.Out, "Release command detected: this new release will not be available until the command succeeds.")
+	}
+
+	fmt.Fprintln(m.io.Out)
+
+	if releaseCommand != nil {
+		// TODO: don't use text block here
+		tb := render.NewTextBlock(ctx, fmt.Sprintf("Release command detected: %s\n", releaseCommand.Command))
+		tb.Done("This release will not be available until the release command succeeds.")
+
+		if err := watch.ReleaseCommand(ctx, m.appCompact.Name, releaseCommand.ID); err != nil {
+			return err
+		}
+
+		release, err = m.apiClient.GetAppReleaseNomad(ctx, m.appCompact.Name, release.ID)
+		if err != nil {
+			return err
+		}
+	}
+	return watch.Deployment(ctx, m.appCompact.Name, release.EvaluationID)
+}
+
+func (m *v2PlatformMigrator) migratePgVolumes(ctx context.Context) error {
+	app := m.appFull
+	regionsToVols := map[string][]api.Volume{}
+	// Find all volumes
+	for _, vol := range app.Volumes.Nodes {
+		if strings.Contains(vol.Name, "machines") {
+			continue
+		}
+		if _, ok := regionsToVols[vol.Region]; ok {
+			regionsToVols[vol.Region] = append(regionsToVols[vol.Region], vol)
+		} else {
+			regionsToVols[vol.Region] = []api.Volume{vol}
+		}
+	}
+
+	var newVols []*NewVolume
+	for region, vols := range regionsToVols {
+		fmt.Fprintf(m.io.Out, "Creatings %d new volume(s) in '%s'", len(vols), region)
+		for _, vol := range vols {
+			// TODO: make use of https://github.com/superfly/nomad-firecracker/pull/1013
+			input := api.CreateVolumeInput{
+				AppID:     app.ID,
+				Name:      fmt.Sprintf("%s_machines", vol.Name),
+				Region:    region,
+				SizeGb:    vol.SizeGb,
+				Encrypted: vol.Encrypted,
+			}
+			if len(vol.Snapshots.Nodes) > 0 {
+				//TODO(dov) figure out if this assumption about ordering is correct
+				input.SnapshotID = &vol.Snapshots.Nodes[len(vol.Snapshots.Nodes)-1].ID
+			}
```

Comment:
is it to ship this with these unresolved?

Reply:
Yeah